src/js/components/scene-editor/media-object-list.jsx

```diff
@@ -89,21 +90,45 @@ var MediaObjectList = React.createClass({
 
     render: function () {
         var items = null;
+        var self =this;
+        var mediaArray = null;
         try {
             if (this.props.scene && this.props.scene.scene && this.props.scene.scene.length !== 0) {
+
+                //AP: attaching tag matcher to the media objects array and plugging in the tagSearch value.
+                var match = _(self.props.scene.scene)
+                    .filter(function(mo) {
+                        return  new TagMatcher("(" + self.state.tagSearch + ")").match(mo.tags);
```

Comment:
Well worth pulling the tag matcher as a variable outside of the foreach, given it's the same each time.

Small performance improvement.